src/azure-cli/azure/cli/command_modules/acs/decorator.py

```diff
@@ -1136,6 +1138,125 @@ def get_admin_username(self, enable_validation: bool = False, **kwargs):
         # this parameter does not need validation
         return admin_username
 
+    # pylint: disable=unused-argument
+    def get_windows_admin_username_and_password(
+        self, enable_validation: bool = False, **kwargs
+    ):
+        # windows_admin_username
+        # read the original value passed by the command
+        username_raw_value = self.raw_param.get("windows_admin_username")
+        # try to read from intermediates
+        username_intermediate = self.get_intermediate(
+            "windows_admin_username", None
+        )
+        # try to read the property value corresponding to the parameter from the `mc` object
+        username_value_obtained_from_mc = None
+        if self.mc and self.mc.windows_profile:
+            username_value_obtained_from_mc = (
+                self.mc.windows_profile.admin_username
+            )
+
+        # set default value
+        username_read_from_mc = False
+        if username_value_obtained_from_mc is not None:
+            windows_admin_username = username_value_obtained_from_mc
+            # clean up intermediate if `mc` has been decorated
+            self.remove_intermediate("windows_admin_username")
```

Comment:
move this removing to the one who modify

Reply:
nice catch, changed in commit [#9f0a0f7](https://github.com/Azure/azure-cli/pull/19351/commits/9f0a0f7798629b0f4a252f9d622a31ba1b04b6c4).